<launch>
	<arg name="marker_size" default="6.4" />
	<arg name="max_new_marker_error" default="0.08" />
	<arg name="max_track_error" default="0.2" />
	
	<arg name="main_cam_image_topic" default="/camera0/image_raw" />
	<arg name="main_cam_info_topic" default="/output/camera_info" />
	<arg name="main_output_frame" default="cam0_optical_frame" />

	<node name="main_ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
		<param name="marker_size"           type="double" value="$(arg marker_size)" />
		<param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
		<param name="max_track_error"       type="double" value="$(arg max_track_error)" />
		<param name="output_frame"          type="string" value="$(arg main_output_frame)" />

		<remap from="camera_image"  to="$(arg main_cam_image_topic)" />
		<remap from="camera_info"   to="$(arg main_cam_info_topic)" />
	</node>

        <arg name="left_cam_image_topic" default="/camera1/image_raw" />
	<arg name="left_cam_info_topic" default="/output/camera_info" />
	<arg name="left_output_frame" default="cam1_optical_frame" />

	<node name="left_ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
		<param name="marker_size"           type="double" value="$(arg marker_size)" />
		<param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
		<param name="max_track_error"       type="double" value="$(arg max_track_error)" />
		<param name="output_frame"          type="string" value="$(arg left_output_frame)" />

		<remap from="camera_image"  to="$(arg left_cam_image_topic)" />
		<remap from="camera_info"   to="$(arg left_cam_info_topic)" />
	</node>

</launch>
